builders/interoperability/xcm/xc-registration/self-serve-asset-registration.md

```diff
@@ -0,0 +1,220 @@
+
+```typescript
+import "@moonbeam-network/api-augment";
+import { ApiPromise, WsProvider } from "@polkadot/api";
+import { blake2AsHex } from "@polkadot/util-crypto";
+
+const moonbeam = await ApiPromise.create({ provider: new WsProvider(MOONBEAM_WSS) });
+
+const tx = moonbeam.tx.evmForeignAssets.createForeignAsset(
+  ASSET_ID,
+  assetLocation,
+  DECIMALS,
+  SYMBOL,
+  NAME
+);
+
+// SCALE-encoded call data (includes call index 0x3800)
+const encodedCall = tx.method.toHex();
+console.log("Encoded call data:", encodedCall);
+
+// Optional: 32-byte call hash (blake2_256)
+console.log("Call hash:", blake2AsHex(encodedCall));
+```
+
+### Dispatch the call with XCM Transact {: #dispatch-the--call-with-xcm-transact }
+
+To register your asset, your parachain must send a three-instruction XCM v4 message that looks like the following instruction set:
+
+```text
+1. WithdrawAsset { assets: (Here, <fee>), effects: [] }
+2. BuyExecution  { fees:   (Here, <fee>), weightLimit: Unlimited }
+3. Transact {
+     originKind: SovereignAccount,
+     requireWeightAtMost: <weight>,
+     call: <encodedCall>
+   }
+```
+
+- `WithdrawAsset` debits the fee from your parachain’s sovereign account on Moonbeam; `Here` means the payment is in GLMR / MOVR / DEV that the account already holds.
+- `BuyExecution` purchases enough weight for the call. You can also use `Unlimited`.
+- `Transact` executes the SCALE-encoded `createForeignAsset` call as your sovereign account (`originKind: SovereignAccount`), with `<weight>` matching or below the weight you bought.
+
+Send the message with `xcmPallet.send` (also known as `polkadotXcm.send`) from your parachain’s root or governance origin, targeting Parachain `2004` for Moonbeam (`2023` for Moonriver):
+
+```rust
+xcmPallet.send(
+    dest: { Parachain: 2004 },
+    message: VersionedXcm::V4(<program above>)
+);
+```
+
+Finally, look for the following event emitted successfully on Moonbeam:
+
+```
+EvmForeignAssets.ForeignAssetCreated(assetId, location, creator)
+```
+
+Its presence confirms the XC-20 asset is live.
+
+### Test First with Chopsticks {: #test-first-with-chopsticks }
+
+Before broadcasting on MainNet, run the flow end-to-end in [Chopsticks](https://github.com/AcalaNetwork/chopsticks){target=\_blank} to verify you have the correct call structure. You can take the following steps in Chopsticks to simulate the call: 
+
+1. Open an HRMP channel between the local parachain and Moonbeam.
+2. Fund the parachain’s sovereign account on the Moonbeam node.
+3. Send the XCM `Transact` from the parachain mock chain.
+4. Observe the `ForeignAssetCreated` event in the Moonbeam logs.
+
+Chopsticks prints full XCM traces, making it easy to verify weight limits, origin matching, and the final event emission.
```

Comment:
Although good practice we don't give enough info to how they can do this. I would scrap this part